plugins/inputs/solr/solr_test.go

```diff
@@ -2,197 +2,198 @@ package solr
+
+			// Load the pages for the test case
+			pages, err := loadPages(testdataPath)
+			require.NoError(t, err)
+
+			// Create a HTTP server that delivers all files in the test-directory
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if !strings.HasPrefix(r.URL.Path, "/solr/") {
+					w.WriteHeader(http.StatusNotFound)
+					return
+				}
+				path := strings.TrimPrefix(r.URL.Path, "/solr/")
+				page, found := pages[path]
+				if !found {
+					w.WriteHeader(http.StatusNotFound)
+					return
+				}
+				w.Write(page)
+			}))
+			require.NotNil(t, server)
+			defer server.Close()
+
+			// Configure the plugin
+			cfg := config.NewConfig()
+			require.NoError(t, cfg.LoadConfig(configFilename))
+			require.Len(t, cfg.Inputs, 1)
+
+			// Setup the plugin
+			plugin := cfg.Inputs[0].Input.(*Solr)
+			plugin.Servers = []string{server.URL}
+			require.NoError(t, plugin.Init())
+
+			// Gather data and compare results
+			var acc testutil.Accumulator
+			require.NoError(t, plugin.Gather(&acc))
+
+			actual := acc.GetTelegrafMetrics()
+			testutil.RequireMetricsEqual(t, expected, actual, options...)
+		})
+	}
 }
 
-func TestSolr7MbeansStats(t *testing.T) {
-	ts := createMockSolr7Server(t)
-	solr := NewSolr()
-	solr.Servers = []string{ts.URL}
-	var acc testutil.Accumulator
-	require.NoError(t, solr.Gather(&acc))
-	acc.AssertContainsTaggedFields(t, "solr_cache",
-		solr7CacheExpected,
-		map[string]string{"core": "main", "handler": "documentCache"})
+func TestIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	// Get all integration test files in testcases
+	resultFiles, err := filepath.Glob(filepath.Join("testcases", "*.result"))
+	require.NoError(t, err)
+
+	// Make sure tests contains data
+	require.NotEmpty(t, resultFiles)
+
+	options := []cmp.Option{
+		testutil.IgnoreTime(),
+		testutil.SortMetrics(),
+	}
+
+	const servicePort = "8983"
+
+	for _, f := range resultFiles {
+		fname := strings.TrimSuffix(filepath.Base(f), ".result")
+		t.Run(fname, func(t *testing.T) {
+			expectedFilename := filepath.Join("testcases", fname+".result")
+
+			// Load the expected output
+			parser := &influx.Parser{}
+			require.NoError(t, parser.Init())
+			expected, err := testutil.ParseMetricsFromFile(expectedFilename, parser)
+			require.NoError(t, err)
+
+			// Determine container version for the integration test
+			// The version number is the last element in the filename separated
+			// by a dash and prefixed with a 'v'.
+			image := "solr"
+			parts := strings.Split(fname, "-")
+			if len(parts) > 1 {
+				version := parts[len(parts)-1]
+				require.True(t, strings.HasPrefix(version, "v"))
+				image += ":" + strings.TrimPrefix(version, "v")
+			}
+
+			// Start the container
+			container := testutil.Container{
+				Image:        image,
+				ExposedPorts: []string{servicePort},
+				Cmd:          []string{"solr-precreate", "main"},
```

Comment:
Is this the command you want the container to start with? Could the existing entrypoint value get used instead?

Reply:
Unfortunately it cannot. You need this command to fill in some data to the container. If you use the usual entry point there is no data and thus some metrics (the important ones we want to test for) are missing...

Reply:
ok thanks for confirming!